src/syft/core/store/store_memory.py

```diff
@@ -63,7 +68,9 @@ def values(self) -> ValuesView[AbstractStorableObject]:
     def store(self, obj: AbstractStorableObject) -> None:
         # TODO: obj should be just "object" and the attributes
         #  of StoreableObject should be put in the metadatastore
-        self._objects[obj.id] = obj
+        with self._lock:
```

Comment:
Another idea to do this is to use the ```cython``` extension and dig into the native ```reference count``` for an object.
*PROS*: it might be faster to directly access that code than also keeping a reference count in the application level
*CONS*: We depend on the ```c``` implementation

*Why I choose this*: Since we do not have yet a benchmarking in place, I think that premature optimization might not be our best choice